File: src/main/java/io/jsql/shutdown/ChannelHander.java
package io.jsql.shutdown;

import io.netty.buffer.Unpooled;
import io.netty.channel.ChannelFuture;
import io.netty.channel.ChannelFutureListener;
import io.netty.channel.ChannelHandlerContext;
import io.netty.channel.ChannelInboundHandlerAdapter;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * Created by 长宏 on 2017/5/4 0004.
 */
public class ChannelHander extends ChannelInboundHandlerAdapter {
    Logger logger = LoggerFactory.getLogger(ChannelHander.class.getName());
    @Override
    public void channelActive(ChannelHandlerContext ctx) throws Exception {
        logger.info("send exit to server......");
        ChannelFuture future = ctx.writeAndFlush(Unpooled.wrappedBuffer(Shutdown.exit_command));
        future.addListener(new ChannelFutureListener() {
            @Override
            public void operationComplete(ChannelFuture future) throws Exception {
                logger.info("send complete...");
                ctx.close();
            }
        });
    }
    @Override
    public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
        super.channelRead(ctx, msg);
    }
    @Override
    public void channelReadComplete(ChannelHandlerContext ctx) throws Exception {
        super.channelReadComplete(ctx);
    }
    @Override
    public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
        ctx.close();
    }
}

File: src/main/java/io/jsql/shutdown/ChannelHander.kt
/*
 * Java-based distributed database like Mysql
 */

package io.jsql.shutdown

import io.netty.buffer.Unpooled
import io.netty.channel.ChannelHandlerContext
import io.netty.channel.ChannelInboundHandlerAdapter
import org.slf4j.LoggerFactory

/**
 * Created by 长宏 on 2017/5/4 0004.
 */
class ChannelHander : ChannelInboundHandlerAdapter() {
    internal var logger = LoggerFactory.getLogger(ChannelHander::class.java.name)
    @Throws(Exception::class)
    override fun channelActive(ctx: ChannelHandlerContext) {
        logger.info("send exit to server......")
        val future = ctx.writeAndFlush(Unpooled.wrappedBuffer(Shutdown.exit_command))
        future.addListener {
            logger.info("send complete...")
            ctx.close()
        }
    }

    @Throws(Exception::class)
    override fun channelRead(ctx: ChannelHandlerContext, msg: Any) {
        super.channelRead(ctx, msg)
    }

    @Throws(Exception::class)
    override fun channelReadComplete(ctx: ChannelHandlerContext) {
        super.channelReadComplete(ctx)
    }

    @Throws(Exception::class)
    override fun exceptionCaught(ctx: ChannelHandlerContext, cause: Throwable) {
        ctx.close()
    }
}
